src/Orleans.Core/Networking/ConnectionManager.cs

```diff
@@ -17,11 +18,12 @@ internal sealed class ConnectionManager
         private static int nextConnection;
 
         private readonly ConcurrentDictionary<SiloAddress, ConnectionEntry> connections = new ConcurrentDictionary<SiloAddress, ConnectionEntry>();
+
         private readonly ConnectionOptions connectionOptions;
         private readonly ConnectionFactory connectionFactory;
         private readonly INetworkingTrace trace;
         private readonly CancellationTokenSource cancellation = new CancellationTokenSource();
-        private readonly object collectionLock = new object();
+        private readonly object lockObj = new object();
 
```

Comment:
There was poor performance/results in recovery tests and I wanted to simplify this class & remove the ~'async spin locks'. So this just uses one regular C# lock now, covering all modifications to `connections` (which is a `ConcurrentDictionary` because we don't lock on reads) and all modifications to the constituent `ConnectionEntry` objects, which are now mutable classes instead of immutable structs, since it simplified the logic (IMO)